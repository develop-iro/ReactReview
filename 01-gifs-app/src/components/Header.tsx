interface HeaderProps {
  title: string;
  description?: string;
}

export const Header = ({ title, description }: HeaderProps) => {
  return (
    <div className="content-center">
      <h1>{title}</h1>
      {description && <p>{description}</p>}
    </div>
  );
};
