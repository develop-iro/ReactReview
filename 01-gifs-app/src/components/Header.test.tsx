import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import { Header } from "./Header.tsx";

describe("Component Header", () => {
  it("matchesÂ snapshot", () => {
    const { asFragment } = render(
      <Header title="Test Title" description="Test Description" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders title and description", () => {
    render(<Header title="Test Title" description="Test Description" />);
    expect(screen.getByText("Test Title")).toBeTruthy();
    expect(screen.getByText("Test Description")).toBeTruthy();
  });
  it("renders title only when description is not provided", () => {
    render(<Header title="Only Title" />);
    expect(screen.getByText("Only Title")).toBeTruthy();
    expect(screen.queryByText("Test Description")).toBeNull();
  });
});
