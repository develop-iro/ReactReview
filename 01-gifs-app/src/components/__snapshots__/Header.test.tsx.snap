// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component Header > matchesÂ snapshot 1`] = `
<DocumentFragment>
  <div
    class="content-center"
  >
    <h1>
      Test Title
    </h1>
    <p>
      Test Description
    </p>
  </div>
</DocumentFragment>
`;
